lib/handoff/transfer_list.c

```diff
@@ -56,35 +66,33 @@ struct transfer_list_header *transfer_list_init(void *p, size_t max_size)
  ******************************************************************************/
 bool transfer_list_relocate(struct transfer_list_header *tl, void *p, size_t max_size)
 {
-	uintptr_t mask = (1 << tl->alignment) - 1;
 	uintptr_t new_tl_base = 0;
 	uintptr_t *np = (uintptr_t *)p;
 
 	if (!tl || !p || max_size == 0)
 		return false;
 
-	new_tl_base = (*np + mask) & ~mask;
+	new_tl_base = ALIGN(*np, tl->alignment);
 
 	if (new_tl_base < *np)
-		new_tl_base += mask + 1;
+		new_tl_base += (1 << tl->alignment);
 
 	if (new_tl_base - *np + tl->size > max_size)
 		return false;
 
-	memmove((void *)new_tl_base, (void *)tl, tl->size);
-	((struct transfer_list_header *)new_tl_base)->max_size = max_size; //to-check
+	memmove((void *)new_tl_base, tl, tl->size);
```

Comment:
you dont need the void here either, it's memove (void *, const void*)).  you could just cast const void* on the second argument

Reply:
Actually the compiler doesn't complain the second argument but the first one, if it is not casted from 'uintptr_t' to 'void *'.

lib/handoff/transfer_list.c:83:10: error: passing argument 1 of ‘memmove’ makes pointer from integer without a cast [-Werror=int-conversion]
```
   83 |  memmove(new_tl_base, tl, tl->size);
        |                    ^~~~~~~~~~~
        |                     |
        |                    uintptr_t {aka long unsigned int}
```

Reply:
Fixed.